extensions/amp-carousel/0.1/build-dom.js

```diff
@@ -0,0 +1,356 @@
+import {isAmp4Email} from '#core/document/format';
+import {isServerRendered} from '#core/dom';
+import {escapeCssSelectorIdent} from '#core/dom/css-selectors';
+import {realChildElements} from '#core/dom/query';
+
+/**
+ * @enum {string}
+ */
+export const ClassNames = {
+  // Carousel Controls
+  BUTTON: 'amp-carousel-button',
+  PREV_BUTTON: 'amp-carousel-button-prev',
+  NEXT_BUTTON: 'amp-carousel-button-next',
+  HAS_CONTROL: 'i-amphtml-carousel-has-controls',
+  CONTROL_HIDE_ATTRIBUTE: 'i-amphtml-carousel-hide-buttons',
+
+  // Generic
+  SLIDE: 'amp-carousel-slide',
+
+  // SlideScroll Carousel
+  SLIDESCROLL_CAROUSEL: 'i-amphtml-slidescroll',
+  SLIDE_WRAPPER: 'i-amphtml-slide-item',
+  SLIDES_CONTAINER: 'i-amphtml-slides-container',
+  SLIDES_CONTAINER_NOSNAP: 'i-amphtml-slidescroll-no-snap',
+
+  // Scrollable Carousel
+  SCROLLABLE_CONTAINER: 'i-amphtml-scrollable-carousel-container',
+  SCROLLABLE_SLIDE: 'amp-scrollable-carousel-slide',
+};
+
+/**
+ * Throws if any provided param is not truthy.
+ * @param  {...any} elements
+ */
+function assertDomQueryResults(...elements) {
+  for (let i = 0; i < elements.length; i++) {
+    if (!elements[i]) {
+      throw new Error('Invalid server render');
+    }
+  }
+}
+
+/**
+ * Builds a carousel button for next/prev.
+ * @param {!Element} element
+ * @param {{className: string, title: string, enabled: boolean}} options
+ * @return {?HTMLDivElement}
+ */
+function buildButton(element, {className, enabled, title}) {
+  /**
+   * In scrollable carousel, the next/previous buttons add no functionality
+   * for screen readers as scrollable carousel is just a horizontally
+   * scrollable div which ATs navigate just like any other content.
+   * To avoid confusion, we therefore set the role to presentation for the
+   * controls in this case.
+   */
```

Comment:
super-nit: This `/** ... */` suggests JSDoc, my inclination would be `//` or at least `/* ... */`

Reply:
Sure